### ISSUE TITLE
Audience URI?

### ISSUE THREAD

**Issue by Syed-SnapLogic** [googleapis/google-auth-library-java#119]
Hi,
  I am trying to check whether a google big query table exists.
  I instantiated Credentials object as follows:

```java
Credentials c = new ServiceAccountJwtAccessCredentials(
                "client ID",
                "client email",
                privateKey,
                "private key id");
        BigQueryOptions bqo = BigQueryOptions
                .newBuilder()
                .setCredentials(c)
                .setProjectId("project ID")
                .build();
        BigQuery bq = bqo.getService();
        TableId id = TableId.of("dataset name", "table name");
        Table t = bq.getTable(id);
        System.out.println("table exists()? : " + t.exists());
```

I am getting the following exception:

```
Exception in thread "main" com.google.cloud.bigquery.BigQueryException: JwtAccess requires Audience uri to be passed in or the defaultAudience to be specified
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.translate(HttpBigQueryRpc.java:86)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.getTable(HttpBigQueryRpc.java:227)
	at com.google.cloud.bigquery.BigQueryImpl$11.call(BigQueryImpl.java:378)
	at com.google.cloud.bigquery.BigQueryImpl$11.call(BigQueryImpl.java:375)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:94)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:54)
	at com.google.cloud.bigquery.BigQueryImpl.getTable(BigQueryImpl.java:375)
	at com.syed.googlecloudstorage.InstantiateStorage.doSecondJob(InstantiateStorage.java:147)
	at com.syed.googlecloudstorage.InstantiateStorage.main(InstantiateStorage.java:131)
Caused by: java.io.IOException: JwtAccess requires Audience uri to be passed in or the defaultAudience to be specified
	at com.google.auth.oauth2.ServiceAccountJwtAccessCredentials.getRequestMetadata(ServiceAccountJwtAccessCredentials.java:257)
	at com.google.auth.http.HttpCredentialsAdapter.initialize(HttpCredentialsAdapter.java:96)
	at com.google.cloud.http.HttpTransportOptions$1.initialize(HttpTransportOptions.java:157)
	at com.google.api.client.http.HttpRequestFactory.buildRequest(HttpRequestFactory.java:93)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.buildHttpRequest(AbstractGoogleClientRequest.java:300)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:419)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.getTable(HttpBigQueryRpc.java:225)
	... 7 more
```

Any clues?  Can somebody help me on what possible values I can supply for audience uri?  Any steps to know that?  Please.

**Comment by elharo** [comment]
So is the bug here or in BigQuery? Do we need to upgrade BigQuery to the latest auth library? Or fix its calls? Or both?

**Comment by chingor13** [comment]
Finally, getting back to this issue - it has 2 parts:

First, is the exception which seems to be a bug and is addressed in #223 - the request initializer is trying to build a token before the request's URI is set. We can delay fetching the token until right before the request is executed using an HttpRequestInterceptor.

Second, is that this use case won't work as the BigQuery API does not allow for signed JWTs - it requires an OAuth 2.0 token so the ServiceAccountJWTCredentials cannot be used against it.

**Comment by sduskis** [comment]
It sounds like there's nothing actionable on this issue in the `google-auth-library-java` repo.  Please contact the BigQuery team for support or a feature request (https://cloud.google.com/bigquery/support) for the signed JWT credentials support.
